### PR TITLE
Fixing one pmem path on AppDirect mode may cause the pmem initialization path to be empty Path

### PR DESCRIPTION
**Pull request by JustDoCoder**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

**Comment by JustDoCoder** [comment]
After many tests, it is ensured that one pmem path is initialized normally without affecting the previous functions

File: src/main/scala/org/apache/spark/storage/BlockManager.scala [hunk 1]
```diff
@@ -194,7 +194,12 @@ private[spark] class BlockManager(
       if (numaNodeId == -1) {
         numaNodeId = executorId.toInt
       }
-      val path = pmemInitialPaths(numaNodeId % 2)
+      var path:String =""
```
Comment (winningsix) [inline src/main/scala/org/apache/spark/storage/BlockManager.scala]: Slightly changed.
```
val path_postfix = File.separator + s"executor_${executorId}" + File.pathSeparator
var file  = if (1 == pmemInitialPaths.size) {
  new File(pmemInitialPaths(0) + path_postfix)
} else {
  new File(pmemInitialPaths(numaNodeId % 2) + path_postfix)
}
```

**Review by winningsix — COMMENTED** [review]
Overall looks good to me.

File: src/main/scala/org/apache/spark/storage/BlockManager.scala (hunk 1 above)
Comment (JustDoCoder) [reply src/main/scala/org/apache/spark/storage/BlockManager.scala]: @winningsix I agree with this change

Comment (JustDoCoder) [reply src/main/scala/org/apache/spark/storage/BlockManager.scala]: @winningsix Please review, thank you.

**Comment by winningsix** [comment]
LGTM

**Pull request merged**